server/__tests__/app.test.js

```diff
@@ -417,15 +417,122 @@ describe('8. GET /api/users', () =>{
   describe('9. GET /api/reviews with queries', () =>{
 
     test("status:200, accepts a category query and should return an array of review object who's category match this value", ()=>{
+        return request(app)
+        .get('/api/reviews?category=social deduction')
+        .expect(200)
+        .then((response)=>{
+            const reviews = response.body.reviews;
+            expect(reviews).toHaveLength(11);
+            reviews.forEach(review =>{
+                expect(review).toEqual(
+                    expect.objectContaining({
+                        comment_count: expect.any(String),
+                        designer: expect.any(String),
+                        owner: expect.any(String),
+                        title: expect.any(String),
+                        review_id: expect.any(Number),
+                        review_img_url: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number)
+                    })
+                )
+                expect(review.category).toBe("social deduction")
+            })  
+            expect(reviews).toBeSortedBy('created_at', {descending: true});
+        })
+    })
+
+    test("status:400, given a bad category query", ()=>{
+        return request(app)
+        .get('/api/reviews?category=asdf')
+        .expect(400)
+        .then((response)=>{
+            expect(response.body.msg).toBe("Category does not exist");
+        })
+    })
+
+    test("status:200, accepts a sort_by query and should return an array of review object sorted based on that query", ()=>{
+        return request(app)
+        .get('/api/reviews?sort_by=designer')
+        .expect(200)
+        .then((response)=>{
+            const reviews = response.body.reviews;
+            expect(reviews).toHaveLength(13);
+            reviews.forEach(review =>{
+                expect(review).toEqual(
+                    expect.objectContaining({
+                        comment_count: expect.any(String),
+                        designer: expect.any(String),
+                        owner: expect.any(String),
+                        title: expect.any(String),
+                        review_id: expect.any(Number),
+                        review_img_url: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        category: expect.any(String)
+                    })
+                )
+            })  
+            expect(reviews).toBeSortedBy('designer', {descending: true});
+        })
+    })
+
+    test("status:400, given a bad sort_by query", ()=>{
+        return request(app)
+        .get('/api/reviews?sort_by=asdf')
+        .expect(400)
+        .then((response)=>{
+            expect(response.body.msg).toBe("Column does not exist");
+        })
```

Comment:
400 for this one is correct as we have full control of what columns we can sort by